index.js

```diff
@@ -12,13 +12,17 @@ async function plugin (app, options) {
     }
 
     const certificate = this.raw.socket.getPeerCertificate(false)
-    const commonName = certificate.subject.CN
+    const commonName = certificate?.subject?.CN
 
```

Comment:
```suggestion
if (!commonName) {
  return {}
}
```

Putting the check here would be enough.

Reply:
Correct. Done.